classes/Kohana/Session.php

```diff
@@ -374,8 +374,14 @@ public function write()
 		catch (Exception $e)
 		{
 			// Log & ignore all errors when a write fails
-			Kohana::$log->add(Log::ERROR, Kohana_Exception::text($e))->write();
+			Kohana::$log->error(Kohana_Exception::text($e));
 
+			if (Kohana::$log instanceof Kohana_Logger)
```

Comment:
You see @rjd22, there is not enough comments to explain the things. I think I need to add some explanation here.